packages/core/src/interfaces/store.ts

```diff
@@ -0,0 +1,219 @@
+import { History, Store } from '@craftjs/utils';
+
+import { EditorEvents } from './editor';
+import { Indicator } from './events';
+import { LegacyStateOptions } from './legacy/editor';
+import { LegacyIndicator } from './legacy/events';
+import {
+  FreshNode,
+  LegacyNode,
+  SerializedNode,
+  SerializedNodes,
+} from './legacy/nodes';
+import {
+  Node,
+  NodeEventTypes,
+  NodeId,
+  NodeRules,
+  NodeSelector,
+  NodeTree,
+} from './nodes';
+
+import { CoreEventHandlers } from '../events';
+import { EditorState, Resolver } from '../interfaces';
+import { RelatedComponents } from '../nodes/RelatedComponents';
+import { ActionMethods } from '../store/actions';
+
+export interface LegacyNodeQuery extends LegacyNode {
+  /**
+   * @deprecated
```

Comment:
*Just using one of those `@deprecated` as an example`

Should the deprecated message contain a link to this PR / a page in the docs / a GitHub issue on how to migrate to the new syntax?

Or maybe print a [`deprecationWarning`](https://github.com/prevwong/craft.js/pull/272/files#diff-aaa1ec60726736a066bf50e9f34d4dc116028a30b488cd347f1a60922c8c953aR229) ?

Reply:
Will create a migration guide once this is a bit more stable